R/opart_gaussian.R

```diff
@@ -37,16 +38,25 @@ opart_gaussian <- structure(function(data, penalty) {
     stop("penalty must be a finite numeric value")
   }
 
-
   result <- .C("opart_gaussian_interface",
-               as.integer(length(data)),
-               as.double(data),
-               as.double(penalty),
-               cost_ptr = as.double(vector("double", length(data))),
-               end_ptr = as.integer(vector("integer", length(data))),
+               n_data = as.integer(length(data)),
+               data.vec = as.double(data),
+               penalty = as.double(penalty),
+               cost.vec = as.double(vector("double", length(data))),
                sums = as.double(vector("double", length(data))),
-               sq_sums = as.double(vector("double", length(data))),
+               dp = as.double(vector("double", length(data))),
+               end.vec = as.integer(vector("integer", length(data))),
+               positions = as.integer(vector("integer", length(data))),
                PACKAGE="opart")
 
-  print(result)
-})
+  seg_ends <- (result$end.vec)
+
+  #remove the -2 from the output
+  result$end.vec <- seg_ends[seg_ends != -2]
+
+  #remove the columns used for internal calculations as they don't need to be displayed
+  result <- result[-c(5,6,8)]
```

Comment:
please refer to objects by name, not by index, when possible

e.g. `result[c("cost.vec", "end.vec")]`